src/API.ts

```diff
@@ -181,6 +185,129 @@ async function getBlob(
   return new Response(rs, { headers }).blob();
 }
 
+let encryptionJobID = 0;
+const decryptionConfigs = new Map<number, PenumbraDecryptionInfo>();
+/**
+ * Get the decryption config for an encrypted file
+ *
+ * ```ts
+ * penumbra.getDecryptionInfo(file: PenumbraEncryptedFile): Promise<PenumbraDecryptionInfo>
+ * ```
+ */
+function getDecryptionInfo(
+  file: PenumbraEncryptedFile,
+): PenumbraDecryptionInfo {
+  if (!decryptionConfigs.has(file.id)) {
+    throw new Error('Unable to find decryption info for file');
+  }
+  return decryptionConfigs.get(file.id) as PenumbraDecryptionInfo;
+}
+
+addEventListener(
+  'penumbra-encryption-complete',
+  ({ detail: { id, decryptionInfo } }) => {
+    console.log(`encryption job #${id} complete`);
+    decryptionConfigs.set(id, decryptionInfo);
+  },
+);
+
+/**
+ * Encrypt files
+ *
+ * ```ts
+ * await penumbra.encrypt(options, ...files);
+ * // usage example:
+ * size = 4096 * 64 * 64;
+ * addEventListener('penumbra-progress',(e)=>console.log(e.type, e.detail));
+ * addEventListener('penumbra-encryption-complete',(e)=>console.log(e.type, e.detail));
+ * file = penumbra.encrypt(null, {stream:intoStream(new Uint8Array(size)), size});
+ * let data = [];
+ * file.then(async ([encrypted]) => {
+ *   console.log('encryption started');
+ *   data.push(new Uint8Array(await new Response(encrypted.stream).arrayBuffer()));
+ * });
+ */
+export async function encrypt(
```

Comment:
This function could use inline comments

Reply:
Sure, I'll try to document the flow from main thread to worker and back.